File: React/exo-rooting-navigate-contactlist/src/model/Person.js
let personCount = 0

class Person {
    static personCount = 0

    constructor(_prenom,_nom,_email,_telephone) {
        this._id = ++personCount
        this._prenom = _prenom
        this._nom = _nom
        this._email = _email
        this._telephone = _telephone
    }

    get id(){
        return this._id
    }

    get prenom(){
        return this._prenom
    }

    get nom(){
        return this._nom
    }

    get email(){
        return this._email
    }

    get telephone(){
        return this._telephone
    }
}

export default Person

File: JAVA/JPA/Demos/demo_jpa/src/main/java/entity/Person.java
package entity;

import javax.persistence.*;

@Entity
@Table(name = "person")
public class Person {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY) // définie la stratégie d'incrémentation - de base en mode auto
    private Long id;


    private String prenom;

    @Column(name = "name") // sert a préciser des informations sur la colonne name
    private String nom;

    @Transient //pas pris en compte dans la base de donnée
    private int age;

    public Person() {
    }

    public Person(String prenom, String nom) {
        this.prenom = prenom;
        this.nom = nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    @Override
    public String toString() {
        return "Person{" +
                "id=" + id +
                ", prenom='" + prenom + '\'' +
                ", nom='" + nom + '\'' +
                ", age=" + age +
                '}';
    }
}
